lxd/db/cluster/identities.go

```diff
@@ -438,6 +438,30 @@ func ActivateTLSIdentity(ctx context.Context, tx *sql.Tx, identifier uuid.UUID,
 	return nil
 }
 
+var getPendingIdentityByTokenSecretStmt = fmt.Sprintf(`
+SELECT identities.id, identities.auth_method, identities.type, identities.identifier, identities.name, identities.metadata
+	FROM identities
+	WHERE identities.type IN (%d, %d)
+	AND json_extract(identities.metadata, '$.secret') = ?
+`, identityTypeCertificateClientPending, identityTypeCertificateClusterLinkPending)
+
+// GetPendingIdentityByTokenSecret gets a single identity and its type, of type identityTypeCertificateClientPending or identityTypeCertificateClusterLinkPending,
+// with the given secret in its metadata. If no pending identity is found, an api.StatusError is returned with http.StatusNotFound.
+func GetPendingIdentityByTokenSecret(ctx context.Context, tx *sql.Tx, secret string) (*Identity, error) {
```

Comment:
Agreed to refactor existing function and use a variadic for the identity types we want (or `nil` for all)